src/detection_win.cpp

```diff
@@ -251,6 +252,38 @@ void extractVidPid(char * buf, ListResultItem_t * item) {
 }
 
 
+void ExtractSerialNumber(char * buf, ListResultItem_t * item) {
+	if (buf == NULL) {
+		return;
+	}
+
+	if (strstr(buf, MI_TAG) != NULL){
+		return; // ignore multiple-interface devices
+	}
+
+	char* temp;
+	char* token;
+	char* string;
+
+	string = new char[strlen(buf) + 1];
+	memcpy(string, buf, strlen(buf) + 1);
+
+	// expected format:
+	// USB\VID_04e8&PID_503b\0002F9A9828E0F06
```

Comment:
Do you have any docs on what this string represents?

Reply:
https://docs.microsoft.com/en-us/windows-hardware/drivers/install/device-ids ->
https://docs.microsoft.com/en-us/windows-hardware/drivers/install/hardware-ids ->
https://docs.microsoft.com/en-us/windows-hardware/drivers/install/device-identifier-formats ->
https://docs.microsoft.com/en-us/windows-hardware/drivers/install/identifiers-for-usb-devices ->

 - [Standard USB Identifiers](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/standard-usb-identifiers)
 - [Special USB Identifiers](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/special-usb-identifiers)

Reply:
> If UniqueID is TRUE, the device instance ID, formed from the bus-supplied device ID and instance ID, uniquely identifies a device in the system.
> 
> https://docs.microsoft.com/en-us/windows-hardware/drivers/install/instance-ids

At least from the devices I have, the only time the `<instance-specific-ID>` looks like an actual serial number is when `CM_DEVCAP_UNIQUEID` is set in the `Capabilities` (looking in Device Manager).

I think we might want to add an if statement to only fill in `serialNumber` if the `UniqueID` capability is set.